pkg/skaffold/verify/k8sjob/verify.go

```diff
@@ -0,0 +1,461 @@
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/util"
+)
+
+// Verifier verifies deployments using kubernetes libs/CLI.
+type Verifier struct {
+	configName string
+
+	cfg              []*latest.KubernetesClusterVerifyTestCase
+	tracker          *tracker.JobTracker
+	imageLoader      loader.ImageLoader
+	logger           *k8sjoblogger.Logger
+	statusMonitor    status.Monitor
+	originalImages   []graph.Artifact // the set of images parsed from the Verifier's manifest set
+	localImages      []graph.Artifact // the set of images marked as "local" by the Runner
+	kubectl          kubectl.CLI
+	labeller         *label.DefaultLabeller
+	defaultNamespace string
+}
+
+// NewVerifier returns a new Verifier for a VerifyConfig filled
+// with the needed configuration for `kubectl apply`
+func NewVerifier(ctx context.Context, cfg kubectl.Config, labeller *label.DefaultLabeller, v *latest.KubernetesClusterVerifier, artifacts []*latest.Artifact) (*Verifier, error) {
+	defaultNamespace := ""
+	b, err := (&util.Commander{}).RunCmdOut(context.Background(), exec.Command("kubectl", "config", "view", "--minify", "-o", "jsonpath='{..namespace}'"))
+	if err == nil {
+		defaultNamespace = strings.Trim(string(b), "'")
+		if defaultNamespace == "default" {
+			defaultNamespace = ""
+		}
+	}
+	if v.DefaultNamespace != nil {
+		var err error
+		defaultNamespace, err = util.ExpandEnvTemplate(*v.DefaultNamespace, nil)
+		if err != nil {
+			return nil, err
+		}
+	}
+	kubectl := kubectl.NewCLI(cfg, v.Flags, defaultNamespace)
+	// default namespace must be "default" not "" when used to create and stream logs from Job(s)
+	if defaultNamespace == "" {
+		defaultNamespace = "default"
+	}
+	tracker := tracker.NewContainerTracker()
+	logger, err := k8sjoblogger.NewLogger(ctx, tracker, labeller, kubectl.KubeContext)
+	if err != nil {
+		return nil, err
+	}
+	var origImages []graph.Artifact
+	for _, artifact := range artifacts {
+		origImages = append(origImages, graph.Artifact{
+			ImageName: artifact.ImageName,
+		})
+	}
+	return &Verifier{
+		originalImages:   origImages,
+		cfg:              v.KubernetesClusterVerifyTestCases,
+		defaultNamespace: defaultNamespace,
+		imageLoader:      component.NewImageLoader(cfg, kubectl.CLI),
+		logger:           logger,
+		statusMonitor:    &status.NoopMonitor{},
+		kubectl:          kubectl,
+		tracker:          tracker,
+		labeller:         labeller,
+	}, nil
+}
+
+func (v *Verifier) ConfigName() string {
+	return v.configName
+}
+
+func (v *Verifier) GetLogger() log.Logger {
+	return v.logger
+}
+
+func (v *Verifier) GetStatusMonitor() status.Monitor {
+	return v.statusMonitor
+}
+
+func (v *Verifier) RegisterLocalImages(images []graph.Artifact) {
+	v.localImages = images
+}
+
+func (v *Verifier) TrackBuildArtifacts(artifacts []graph.Artifact) {
+	v.logger.RegisterArtifacts(artifacts)
+}
+
+// Verify executes specified artifacts by creating containers in the local docker daemon
```

Comment:
This looks like the description for the docker verifier, here it is a k8s job, not a container in the local docker daemon.

Reply:
done